boletin-de-pediatria.csl

```diff
@@ -18,109 +18,140 @@
   </info>
   <macro name="author">
-    <names variable="author" suffix=". ">
+    <names variable="author">
       <name sort-separator=" " initialize-with="" name-as-sort-order="all" delimiter=", " delimiter-precedes-last="always"/>
-      <label form="short" prefix=" (" suffix=")" strip-periods="true"/>
+      <label form="long" prefix=", "/>
       <substitute>
         <names variable="editor"/>
       </substitute>
     </names>
   </macro>
   <macro name="editor">
-    <names variable="editor">
+    <names variable="editor" suffix=".">
       <name sort-separator=" " initialize-with="" name-as-sort-order="all" delimiter=", " delimiter-precedes-last="always"/>
-      <label form="short" prefix=" (" suffix=")" strip-periods="true"/>
+      <label form="long" prefix=", "/>
     </names>
   </macro>
-  <macro name="date">
+  <macro name="chapter-marker">
     <choose>
-      <if type="article-newspaper report" match="any">
-        <date variable="issued" form="text"/>
+      <if type="chapter paper-conference entry-dictionary entry-encyclopedia" match="any">
+        <text term="in" text-case="capitalize-first"/>
       </if>
-      <else>
-        <date variable="issued" form="text" date-parts="year"/>
-      </else>
     </choose>
   </macro>
   <macro name="publisher">
-    <group delimiter=", ">
-      <group delimiter=" ">
-        <text variable="genre" text-case="title"/>
-        <text variable="number"/>
-      </group>
-      <group delimiter=": ">
-        <choose>
-          <if type="thesis" match="none">
-            <text variable="publisher-place"/>
-          </if>
-        </choose>
-        <text variable="publisher"/>
-      </group>
-      <choose>
-        <if variable="URL" match="none">
-          <text macro="date"/>
-        </if>
-      </choose>
-    </group>
+    <choose>
+      <!--discard publisher info for articles-->
+      <if type="article-journal article-magazine article-newspaper" match="none">
+        <group delimiter=": " suffix=";">
+          <choose>
+            <if type="thesis">
+              <text variable="publisher-place" prefix="[" suffix="]"/>
+            </if>
+            <else-if type="speech"/>
+            <else>
+              <text variable="publisher-place"/>
+            </else>
+          </choose>
+          <text variable="publisher"/>
+        </group>
+      </if>
+    </choose>
   </macro>
   <macro name="access">
     <choose>
-      <if variable="page" match="none">
-        <choose>
-          <if variable="DOI" match="any">
-            <group prefix=". " delimiter=". ">
-              <date variable="issued" form="text" prefix="Epub ahead of print "/>
-              <text variable="DOI" prefix="DOI: "/>
-            </group>
-          </if>
-          <else>
-            <group delimiter=" ">
-              <text variable="URL"/>
-              <text macro="accessed-date"/>
-            </group>
-          </else>
-        </choose>
+      <if variable="URL">
```

Comment:
Please use logic like above to prefer DOI over URL and use a prefix "https://doi.org/"

Reply:
Thanks for the suggestion. I'll upload some changes in this macro.

Reply:
I suggest:

```
  <macro name="access">
    <choose>
      <if variable="page" match="none">
        <choose>
	 <if variable="DOI" match="any">
            <group delimiter=". ">
              <text variable="DOI" prefix="DOI: https://doi.org/"/>
            </group>
          </if>
          <else>
            <group delimiter=" ">
	      <text term="available at" text-case="capitalize-first" suffix=": "/>
              <text variable="URL" />
            </group>
          </else>
        </choose>
	</if>
    </choose>
  </macro>
```

Reply:
Use 

```
  <macro name="access">
    <choose>
      <if variable="page" match="none">
        <choose>
	 <if variable="DOI" match="any">
            <text variable="DOI" prefix="https://doi.org/"/>
          </if>
          <else>
            <group delimiter=" ">
	      <text term="available at" text-case="capitalize-first" suffix=": "/>
              <text variable="URL" />
            </group>
          </else>
        </choose>
	</if>
    </choose>
  </macro>
```

Reply:
OK